runtime/server/queries_metrics.go

```diff
@@ -484,6 +489,67 @@ func (s *Server) MetricsViewTimeRanges(ctx context.Context, req *runtimev1.Metri
 	}, nil
 }
 
+func (s *Server) MetricsViewAnnotations(ctx context.Context, req *runtimev1.MetricsViewAnnotationsRequest) (*runtimev1.MetricsViewAnnotationsResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.instance_id", req.InstanceId),
+		attribute.String("args.metric_view", req.MetricsViewName),
+		attribute.String("args.annotation", req.AnnotationName),
+	)
+
+	if !auth.GetClaims(ctx).CanInstance(req.InstanceId, auth.ReadMetrics) {
+		return nil, ErrForbidden
+	}
+	s.addInstanceRequestAttributes(ctx, req.InstanceId)
+
+	res, err := s.runtime.Resolve(ctx, &runtime.ResolveOptions{
+		InstanceID: req.InstanceId,
+		Resolver:   "annotations",
+		ResolverProperties: map[string]any{
+			"metrics_view": req.MetricsViewName,
+			"annotation":   req.AnnotationName,
+		},
+		Args: map[string]any{
+			"priority": req.Priority,
+			"time_range": metricsview.TimeRange{
+				Start:         req.TimeRange.Start.AsTime(),
+				End:           req.TimeRange.End.AsTime(),
+				Expression:    req.TimeRange.Expression,
+				IsoDuration:   req.TimeRange.IsoDuration,
+				IsoOffset:     req.TimeRange.IsoOffset,
+				RoundToGrain:  metricsview.TimeGrainFromProto(req.TimeRange.RoundToGrain),
+				TimeDimension: req.TimeRange.TimeDimension,
+			},
+			"time_grain": req.TimeGrain,
+		},
+		Claims: auth.GetClaims(ctx).SecurityClaims(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer res.Close()
+
+	data := make([]*structpb.Struct, 0)
+	for {
+		row, err := res.Next()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return nil, err
+		}
+		rowStruct, err := pbutil.ToStruct(row, res.Schema())
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		data = append(data, rowStruct)
+	}
```

Comment:
Mentioned this earlier, but would it make sense to parse into a more strictly typed response proto here? Or do you prefer it like this?